src/NServiceBus.Core/Unicast/RunningEndpointInstance.cs

```diff
@@ -63,15 +65,23 @@ public async Task Stop(CancellationToken cancellationToken = default)
                 finally
                 {
                     settings.Clear();
-                    (builder as IDisposable)?.Dispose();
+                    if (builder is IAsyncDisposable asyncDisposableBuilder)
```

Comment:
@danielmarbach maybe we should add a comment there with a reference to https://github.com/Particular/NServiceBus/blob/master/src/NServiceBus.Core/Endpoint.cs#L23-L25 as @abparticular would have been my same comment.
```suggestion
					// No need to also support `IDisposable`, only invoked for internally managed mode. See `Endpoint.Create`
                    if (builder is IAsyncDisposable asyncDisposableBuilder)
```

Reply:
I've pushed some changes to make things more revealing (I hope)